src/paulscode/android/mupen64plusae/profile/TouchscreenProfileActivity.java

```diff
@@ -385,37 +397,125 @@ private void hideSystemBars()
+                }
+            }
+            
+            dragX = mProfile.getInt( dragAsset + TAG_X, INITIAL_ASSET_POS );
+            dragY = mProfile.getInt( dragAsset + TAG_Y, INITIAL_ASSET_POS );
+            
+            return true;
         }
-        else
+        else if( ( event.getAction() & MotionEvent.ACTION_MASK ) == MotionEvent.ACTION_MOVE )
         {
-            // See if analog was pressed
-            Point point = mTouchscreenMap.getAnalogDisplacement( x, y );
-            int dX = point.x;
-            int dY = point.y;
-            float displacement = FloatMath.sqrt( ( dX * dX ) + ( dY * dY ) );
-            if( mTouchscreenMap.isInCaptureRange( displacement ) )
-                popupDialog( ANALOG, getString( R.string.controller_analog ), -1 );
+            if ( dragIndex != TouchMap.UNMAPPED || ANALOG.equals(dragAsset) )
+            {
+                if ( !dragging )
+                {
+                    int dX = x - initialX;
+                    int dY = y - initialY;
+                    float displacement = FloatMath.sqrt( ( dX * dX ) + ( dY * dY ) );
+                    if ( displacement >= 10 )
+                        dragging = true;
+                }
+                if ( !dragging )
+                    return false;
+                
+                // drag this button or analog stick around
+                
+                // calculate the X and Y percentage
+                View view = getWindow().getDecorView();
+                int newDragX = ( x - ( initialX - dragFrame.left ) ) * 100/( view.getWidth() - ( dragFrame.right - dragFrame.left ) );
+                int newDragY = ( y - ( initialY - dragFrame.top ) ) * 100/( view.getHeight() - ( dragFrame.bottom - dragFrame.top ) );
+                
+                // round to the nearest 5%
+                newDragX = Math.min( Math.max( Math.round( newDragX / 5 ) * 5, 0 ), 100 );
+                newDragY = Math.min( Math.max( Math.round( newDragY / 5 ) * 5, 0 ), 100 );
+                
+                if ( newDragX != dragX || newDragY != dragY )
+                {
+                    dragX = newDragX;
+                    dragY = newDragY;
+                    mProfile.put( dragAsset + TAG_X, String.valueOf( newDragX ) );
+                    mProfile.put( dragAsset + TAG_Y, String.valueOf( newDragY ) );
+                    refresh();
+                }
+            }
+        }
+        else if( ( event.getAction() & MotionEvent.ACTION_MASK ) == MotionEvent.ACTION_UP )
+        {
+            // if this touch was part of a drag/swipe gesture then don't tap the button
+            if ( dragging )
+                return false;
+            
+            // show the editor for the tapped button
+            if ( ANALOG.equals( dragAsset ) )
+            {
+                // play the standard button sound effect
+                View view = getWindow().getDecorView();
+                view.playSoundEffect( SoundEffectConstants.CLICK );
```

Comment:
From what I can gather online, this is automatically disabled by the OS if view sound effects are disabled globally. So it shouldn't need to check for any preferences here.